src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -473,10 +473,13 @@ static bool should_use_ares(const char* resolver_env) {
 }
 #endif /* GRPC_UV */
 
+static bool g_grpc_resolver_dns_ares_initialized;
+
 void grpc_resolver_dns_ares_init() {
   grpc_core::UniquePtr<char> resolver =
       GPR_GLOBAL_CONFIG_GET(grpc_dns_resolver);
   if (should_use_ares(resolver.get())) {
+    g_grpc_resolver_dns_ares_initialized = true;
     gpr_log(GPR_DEBUG, "Using ares dns resolver");
     address_sorting_init();
     grpc_error* error = grpc_ares_init();
```

Comment:
doesn't look like this will be called multiple times. If g_grpc_resolver_dns_ares_initialized is set to true in the previous grpc_init - grpc_shutdown phase, it will remain true and grpc_ares_init will not be called in the second grpc_init - grpc_shutdown phase

Reply:
I updated the description of PR to show an example of best practice. grpc_resolver_dns_ares_init will be called every time grpc_init is called. So any change that user makes before calling grpc_init can be applied to this.